gcloud/_helpers.py

```diff
@@ -322,6 +322,49 @@ def _to_bytes(value, encoding='ascii'):
         raise TypeError('%r could not be converted to bytes' % (value,))
 
 
+def _has_field(message_pb, property_name):
+    """Determine if a field is set on a protobuf.
+
+    :type message_pb: :class:`google.protobuf.message.Message`
+    :param message_pb: The message to check for ``property_name``.
+
+    :type property_name: str
+    :param property_name: The property value to check against.
+
+    :rtype: bool
+    :returns: Flag indicating if ``property_name`` is set on ``message_pb``.
+    """
+    # NOTE: As of proto3, HasField() only works for message fields, not for
+    #       singular (non-message) fields. First try to use HasField and
+    #       if it fails (with a ValueError) we manually consult the fields.
```

Comment:
Let me try to illustrate my intent with an example.

```python
>>> C = gcloud_bigtable._generated.bigtable_cluster_data_pb2.Cluster()
>>> C._fields
{}
>>> C.serve_nodes
0
>>> C._fields
{}
>>> C.serve_nodes = 1
>>> C._fields
{<google.protobuf.descriptor.FieldDescriptor object at 0x7fef469f0450>: 1}
>>> C.HasField('serve_nodes')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/dist-packages/google/protobuf/internal/python_message.py", line 767, in HasField
    raise ValueError(error_msg % field_name)
ValueError: Protocol message has no non-repeated submessage field "serve_nodes"
>>> C.serve_nodes = 0
>>> C._fields
{}
>>> C.name
u''
>>> C.name = u'hi'
>>> C._fields
{<google.protobuf.descriptor.FieldDescriptor object at 0x7fef469f0350>: u'hi'}
>>> C.name = u''
>>> C._fields
{}
>>> C.delete_time
<google.protobuf.timestamp_pb2.Timestamp object at 0x7fef44ec3230>
>>> C._fields
{<google.protobuf.descriptor.FieldDescriptor object at 0x7fef469f0390>: <google.protobuf.timestamp_pb2.Timestamp object at 0x7fef44ec3230>}
>>> C.HasField('delete_time')
False
>>> from gcloud_bigtable._generated.timestamp_pb2 import Timestamp
>>> C.delete_time.CopyFrom(Timestamp())
>>> C._fields
{<google.protobuf.descriptor.FieldDescriptor object at 0x7fef469f0390>: <google.protobuf.timestamp_pb2.Timestamp object at 0x7fef44ec3230>}
>>> C.HasField('delete_time')
True
```

----

I'm kind of realizing that we should just check for truthiness for simple fields.